text/0374-jsii-ts-version.md

```diff
@@ -0,0 +1,656 @@
+package in the toolchain (those are unaffected by this change).
+
+The `jsii-rosetta` package also internally uses the [TypeScript] compiler that
+needs to be compatible with the [TypeScript] language level used by the project.
+Developers hence need to also be able to control which [TypeScript] version is
+being used by `jsii-rosetta` if they want to use this.
+
+Using the same `major.minor` version as the [TypeScript] compiler makes it easy
+for developers to identify which [TypeScript] language features are available to
+them or not. As the [TypeScript] compiler does not conform to [semantic
+versioning][semver], this implies `jsii` and `jsii-rosetta` also need to stop
+conforming to it.
+
+### Why should we _not_ do this?
+
+[Semantic versioning][semver] is the dominant versioning scheme in the
+JavaScript world, and diverging from it might break assumptions customers make
+when consuming packages from [npm](npmjs.com).
+
+Following [TypeScript] versions means releasing new `major.minor` releases (that
+may include breaking changes) more often than is currently done (the
+[TypeScript] maintainers typically declare a new `major.minor` line four
+times per year: in February, May, August and November respectively). This is a
+lot more release lines than other AWS products offer, and it would not be
+reasonable to uphold the [AWS SDKs and Tools maintenance policy][aws-policy],
+as it requires offering full support (features and bug fixes) for "Generally
+Available" releases for a minimum of 24 months (this would require providing
+full support for 8 different releases, and back-porting automation is likely to
+be difficult due to the amount of breaking changes in the [TypeScript] compiler
+API between releases). Diverging from this [policy][aws-policy] is a significant
+change that customers may not expect.
+
+Additionally, since we cannot afford to back-port new features (especially if
+significant) on older releases, this means users may need to update their code
+in order to gain access to new features introduced in the compiler. While this
+may be a desirable forcing function on developers to upgrade to the latest
+release line, it may impose challenges when such features cannot be implemented
+in a way that is compatible with previous versions of the compiler (e.g: it may
+be difficult or impossible to implement [RFC 193][RFC193] in a backwards
+compatible manner, as it likely requires the introduction of a new type kind,
+which previous release lines would not recognize).
+
+[RFC193]: https://github.com/aws/aws-cdk-rfcs/pull/194
+
+### What is the technical solution (design) of this feature?
+
+The proposed delivery plan for this feature is as follows:
+
+* Communicate about the upcoming change in new releases on the `v1` release
+  line, including a planned timeline for the initial release of the `v4.9` line,
+  and language about the departure from [SemVer].
+
+* **Optional:** Separate the `jsii` compiler from the rest of packages in the
```

Comment:
This is worthwhile. I think it's confusing if `jsii` follows different versioning than `jsii-pacmak` when they live in the same repo.